nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/DynamicReloader.java

```diff
@@ -305,6 +324,16 @@ private boolean needsReload() {
             return answer;
         }
         
+        private Properties readReloadFile() {
+            Properties prop = new Properties();
+            try {
+                prop.load(new FileInputStream(reloadFile));
+            } catch (Exception ex) {
+                // skip
+            }
```

Comment:
The input stream should be closed. The [Javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html#load-java.io.InputStream-) states: "The specified stream remains open after this method returns."
```suggestion
            try (InputStream istream = new FileInputStream(reloadFile) {
                prop.load(istream);
            } catch (Exception ex) {
                // skip
            }
```